datalad.distribution.tests.test_get failures on real win10/conda

The appveyor environment is supercharged in many ways. Here is a glimpse of reality starting from https://github.com/datalad-handbook/book/issues/132:

```
> python -m nose datalad.distribution.tests.test_get
======================================================================
FAIL: datalad.distribution.tests.test_get.test_get_recurse_dirs
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:\users\mih\code\datalad\datalad\tests\utils.py", line 1228, in assert_status
    assert_in(r['status'], label)
  File "C:\Users\mih\Miniconda3\lib\unittest\case.py", line 1106, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "C:\Users\mih\Miniconda3\lib\unittest\case.py", line 680, in fail
    raise self.failureException(msg)
AssertionError: 'notneeded' not found in ['ok']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\mih\Miniconda3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "c:\users\mih\code\datalad\datalad\tests\utils.py", line 434, in newfunc
    return t(*(arg + (d,)), **kw)
  File "c:\users\mih\code\datalad\datalad\tests\utils.py", line 607, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "c:\users\mih\code\datalad\datalad\distribution\tests\test_get.py", line 229, in test_get_recurse_dirs
    assert_status('ok', result)
  File "c:\users\mih\code\datalad\datalad\tests\utils.py", line 1234, in assert_status
    dumps(results, indent=1, default=lambda x: str(x))))
AssertionError: Test 4/4: expected status ['ok'] not found in:
[
 {
  "type": "file",
  "refds": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0",
  "status": "ok",
  "path": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0\\subdir/subsubdir/file4.txt",
  "action": "get",
  "annexkey": "MD5E-s9--437b930db84b8079c2dd804a71936b5f.txt",
  "message": "from origin..."
 },
 {
  "type": "file",
  "refds": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0",
  "status": "ok",
  "path": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0\\subdir/file2.txt",
  "action": "get",
  "annexkey": "MD5E-s14--6c7ba9c5a141421e1c03cb9807c97c74.txt",
  "message": "from origin..."
 },
 {
  "type": "file",
  "refds": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0",
  "status": "ok",
  "path": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0\\subdir/subsubdir/file3.txt",
  "action": "get",
  "annexkey": "MD5E-s30--c22e53b13855f9e313a02bcb7c26a7cc.txt",
  "message": "from origin..."
 },
 {
  "action": "get",
  "path": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0\\subdir",
  "type": "directory",
  "refds": "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0",
  "status": "notneeded",
  "message": [
   "nothing to get from %s",
   "C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_n06xb8w0\\subdir"
  ]
 }
]

======================================================================
FAIL: datalad.distribution.tests.test_get.test_get_flexible_source_candidates_for_submodule
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\mih\Miniconda3\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "c:\users\mih\code\datalad\datalad\tests\utils.py", line 607, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "c:\users\mih\code\datalad\datalad\tests\utils.py", line 607, in newfunc
    return t(*(arg + (filename,)), **kw)
  File "c:\users\mih\code\datalad\datalad\distribution\tests\test_get.py", line 87, in test_get_flexible_source_candidates_for_submodule
    eq_(f(clone, 'sub'), [t + '/sub'])
AssertionError: ['C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_et9vdsui\\sub'] != ['C:\\Users\\mih\\AppData\\Local\\Temp\\datalad_temp_et9vdsui/sub']

----------------------------------------------------------------------
Ran 13 tests in 241.061s

FAILED (SKIP=5, failures=2)
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -91,8 +91,8 @@ test_script:
   - "python -m nose -s -v --with-cov --cover-package datalad datalad.cmdline"
   # remaining fails: test_archives.test_basic_scenario test_datalad.test_basic_scenario_local_url
   #- python -m nose -s -v datalad.customremotes
-  # remaining fails: test_add test_create_test_dataset test_get test_uninstall test_utils
-  - "python -m nose -s -v --with-cov --cover-package datalad datalad.distribution.tests.test_create_github datalad.distribution.tests.test_dataset_binding datalad.distribution.tests.test_siblings datalad.distribution.tests.test_update datalad.distribution.tests.test_dataset datalad.distribution.tests.test_publish datalad.local.tests.test_subdataset datalad.distribution.tests.test_clone datalad.distribution.tests.test_create_sibling"
+  # remaining fails: test_add test_create_test_dataset test_uninstall test_utils
+  - "python -m nose -s -v --with-cov --cover-package datalad datalad.distribution.tests.test_get datalad.distribution.tests.test_create_github datalad.distribution.tests.test_dataset_binding datalad.distribution.tests.test_siblings datalad.distribution.tests.test_update datalad.distribution.tests.test_dataset datalad.distribution.tests.test_publish datalad.local.tests.test_subdataset datalad.distribution.tests.test_clone datalad.distribution.tests.test_create_sibling"
   # remaining fails: test_http
   - "python -m nose -s -v --with-cov --cover-package datalad datalad.downloaders.tests.test_credentials datalad.downloaders.tests.test_providers datalad.downloaders.tests.test_s3"
   # remaining fails: test_add_archive_content test_annotate_paths test_diff test_ls_webui test_run test_save test_utils

--- a/datalad/distribution/tests/test_get.py
+++ b/datalad/distribution/tests/test_get.py
@@ -12,6 +12,7 @@
 
 
 from os import curdir
+import os.path as op
 from os.path import join as opj, basename
 from glob import glob
 
@@ -84,10 +85,11 @@ def test_get_flexible_source_candidates_for_submodule(t, t2):
 
     # but if we work on dsclone then it should also add urls deduced from its
     # own location default remote for current branch
-    eq_(f(clone, 'sub'), [t + '/sub'])
-    eq_(f(clone, 'sub', sshurl), [t + '/sub', sshurl])
-    eq_(f(clone, 'sub', httpurl), [t + '/sub'] + sm_httpurls)
-    eq_(f(clone, 'sub'), [t + '/sub'])  # otherwise really we have no clue were to get from
+    subpath = op.sep.join((t, 'sub'))
+    eq_(f(clone, 'sub'), [subpath])
+    eq_(f(clone, 'sub', sshurl), [subpath, sshurl])
+    eq_(f(clone, 'sub', httpurl), [subpath] + sm_httpurls)
+    eq_(f(clone, 'sub'), [subpath])  # otherwise really we have no clue were to get from
     # TODO: check that http:// urls for the dataset itself get resolved
 
     # TODO: many more!!

--- a/datalad/interface/results.py
+++ b/datalad/interface/results.py
@@ -20,9 +20,12 @@ from os.path import join as opj
 from os.path import relpath
 from os.path import abspath
 from os.path import normpath
-from datalad.utils import assure_list
-from datalad.utils import with_pathsep as _with_sep
-from datalad.utils import path_is_subpath
+from datalad.utils import (
+    assure_list,
+    with_pathsep as _with_sep,
+    path_is_subpath,
+    PurePosixPath,
+)
 from datalad.support.path import robust_abspath
 
 from datalad.distribution.dataset import Dataset
@@ -214,7 +217,7 @@ def annexjson2result(d, ds, **kwargs):
     # we cannot rely on any of these to be available as the feed from
     # git annex (or its wrapper) is not always homogeneous
     if 'file' in d:
-        res['path'] = opj(ds.path, d['file'])
+        res['path'] = str(ds.pathobj / PurePosixPath(d['file']))
     if 'command' in d:
         res['action'] = d['command']
     if 'key' in d:

